From 265cd6a5c1ecf7e811c2d18d2588b88264967220 Mon Sep 17 00:00:00 2001
From: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Date: Wed, 8 Dec 2021 23:58:59 +0200
Subject: [PATCH 14/14] plat: s32g: Use -Os optimization level on release
 builds

Use different optimization levels for debug and release builds (DEBUG=1
vs DEBUG=0).

Upstream-Status: Pending 

Signed-off-by: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
Issue: ALB-8269
---
 plat/nxp/s32/s32g/s32g_common.mk | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/plat/nxp/s32/s32g/s32g_common.mk b/plat/nxp/s32/s32g/s32g_common.mk
index 9dd10be1a..09f65453a 100644
--- a/plat/nxp/s32/s32g/s32g_common.mk
+++ b/plat/nxp/s32/s32g/s32g_common.mk
@@ -257,7 +257,11 @@ BL2_EL3_STACK_ALIGNMENT :=	512
 $(eval $(call add_define_val,BL2_EL3_STACK_ALIGNMENT,$(BL2_EL3_STACK_ALIGNMENT)))
 
 ### Devel & Debug options ###
-CFLAGS			+= -O0
+ifeq (${DEBUG},1)
+	CFLAGS			+= -O0
+else
+	CFLAGS			+= -Os
+endif
 # Enable dump of processor register state upon exceptions while running BL31
 CRASH_REPORTING		:= 1
 # As verbose as it can be
-- 
2.17.1

